types/chrome/test/index.ts

```diff
@@ -696,3 +696,200 @@ function testSetBrowserBadgeText() {
     chrome.browserAction.setBadgeText(); // $ExpectError
     chrome.browserAction.setBadgeText(undefined); // $ExpectError
 }
+
+// https://developer.chrome.com/docs/extensions/reference/action/
+async function testActionForPromise() {
+    await chrome.action.disable(0);
```

Comment:
Does `await` actually tests if the function is a `Promise`? I believe it will work even if the function is not a `Promise`.

[Documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)

>[rv] = await expression

>expression
A Promise or any value to wait for.

>rv
Returns the fulfilled value of the promise, or the value itself if it's not a Promise.

Proper way to test it should be with `.then()`.

Reply:
@henev The documentation you mentioned is for JavaScript, not for Typescript, right?

The TSLint is used for the test of the `types/chrome/test/index.ts` file. And, the value of the `await-promise` option is `true`. That is, the `await` can't be specified for non-promise values.

For example, when updating the line above to `await chrome.action.disable(0, () => {});` (there is no definition which returns `Promise` value against the method with a callback function as 2nd argument) and run the test, the following error occurs and the test is failed:

```
Error: /home/yoichiro/projects/typescript/DefinitelyTyped/types/chrome/test/index.ts:702:5
ERROR: 702:5  await-promise  Invalid 'await' of a non-Promise value.
```

As the result, the test code set is valid, I think.

Reply:
I see. Thank you for clarification.